### PR TITLE
fix PHP7.4 error due to wrong config implementation in EmailFactory

### LINKED ISSUE #19: bolt-boltforms.yml config vars aren't read correctly
**bolt 4.1 + bolt/forms**

When submitting a form it will crash with the error: `An exception has been thrown during the rendering of a template ("Notice: Trying to access array offset on value of type null").`

When dumping  the config var in `vendor/bolt/forms/src/factory/EmailFactory.php` at line 29 like so
```
dd($config->get('debug'));
$debug = (bool) $config->get('debug')['enabled']; 
```
an error is triggered because the config is null

Screenshot 2020-10-16 at 13 26 07

### PR DESCRIPTION
**Pull request by LordSimal**
Fixes #19

## What was the problem

Basically with PHP 7.4 after sending a form there has been an error that `Trying to access array offset on value of type null` appeared in `src/factory/EmailFactory.php` at Line 29

After some debugging I found out, that the given `$config` variable was not the collection of the whole `config/extensions/bolt-boltforms.yaml` but rather just the subset of the currently used form.

So like just this subset: (and ongoing till the fields of this form end)
image

But the logic in `src/factory/EmailFactory.php` used config keys like `debug` and `address` which (partially) aren't present in this subset.

So therefore there was a mix between config for the form and the "whole" config for the extension.

## What I did

Basically I went into `src/EventSubscriber/Mailer.php` and added the `$this->event->getConfig()` as a parameter after the already present `$this->event->getFormConfig()`

After that adjusted the parameters for the `create()` function in `src/Factory/EmailFactory.php` to match the new parameters and fixed the mixed config variables.

So sometimes we want to access config just for the current form, but sometimes we want to access the config for the "global" extension.

This PR should fix this.

I `dd()` the variables which previously caused problems and they are now correct in my opinion.

### CODE REVIEW

**Comment by iamgoodbytes** [comment]
Just had the chance to test this and it works on PHP7.4 (CentOS). Thanks for the fix @LordSimal

**Comment by I-Valchev** [comment]
@LordSimal Thanks for looking into this and fixing it. Merging it in right away!
